### PR TITLE
[squeezebox] Support for more audio streams through the audio servlet

### PR DESCRIPTION
**Pull request by dalgwen**
Squeezebox audio sink supporting more audio streams, with the help of the HTTP audio servlet new capabilities.

Related to openhab/openhab-addons#15113

### CODE REVIEW

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java [hunk 1]
```diff
@@ -81,43 +76,62 @@ public String getLabel(Locale locale) {
...
+        }
         AudioFormat format = audioStream.getFormat();
         if (!AudioFormat.WAV.isCompatible(format) && !AudioFormat.MP3.isCompatible(format)) {
+            tryClose(audioStream);
             throw new UnsupportedAudioFormatException("Currently only MP3 and WAV formats are supported: ", format);
         }
 
         String url;
         if (audioStream instanceof URLAudioStream) {
             url = ((URLAudioStream) audioStream).getURL();
```
Comment (lolodomo) [inline bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java:92]: Call tryClose.

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java [hunk 2]
```diff
@@ -81,43 +76,62 @@ public String getLabel(Locale locale) {
...
+                    url += AudioStreamUtils.EXTENSION_SEPARATOR + FileAudioStream.MP3_EXTENSION;
+                }
+
+                // Form the URL for streaming the notification from the OH web server
+                // Use the callback URL if it is set in the binding configuration
+                String host = callbackUrl == null || callbackUrl.isEmpty() ? playerHandler.getHostAndPort()
+                        : callbackUrl;
+                if (host == null) {
+                    logger.warn("Unable to get host/port from which to stream notification");
+                    return;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java:111]: Call tryClose.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java (hunk 1 above)
Comment (dalgwen) [reply bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java:92]: Décidément !
Done.

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java (hunk 2 above)
Comment (dalgwen) [reply bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java:111]: Done

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**